astroid/bases.py

```diff
@@ -436,7 +436,14 @@ def _infer_builtin_new(
         node_context = context.extra_context.get(caller.args[0])
         infer = caller.args[0].infer(context=node_context)
 
-        yield from (Instance(x) if x is not Uninferable else x for x in infer)  # type: ignore[misc,arg-type]
+        for inferred in infer:
+            if isinstance(inferred, type(Uninferable)):  # isinstance helps mypy
```

Comment:
```suggestion
            if inferred is Uninferable:
```

Let's fix all those `is Uninferable` statements with #1680.

Reply:
Are you suggesting to block this until #1680? That could be a long way off. Or just to silence the mypy error?

Reply:
Ah, right, we're not running mypy. So just letting it fail, then?

Reply:
let if fail 😄